js/viz/gauges/utils/resolve_label_overlapping.js

```diff
@@ -0,0 +1,104 @@
+const ARC_COORD_PREC = 5;
+
+import { normalizeAngle, normalizeArcParams } from '../../core/utils';
+
+export function dividePoints(bars) {
+    return bars.reduce(function(r, bar) {
+        const angle = normalizeAngle(bar._angle);
+        bar._text._lastCoords = { x: 0, y: 0 };
+        (angle <= 90 || angle >= 270 ? r.right : r.left)
+            .push({
+                series: {
+                    isStackedSeries: ()=> false,
+                    isFullStackedSeries: ()=> false
+                },
+                getLabels: ()=> [{
+                    isVisible: ()=> true,
+                    getBoundingRect: () => {
+                        const { height, width, x, y } = bar._text.getBBox();
+                        const lastCoords = bar._text._lastCoords;
+
+                        return {
+                            x: x + lastCoords.x,
+                            y: y + lastCoords.y,
+                            width: width,
+                            height: height,
+                        };
+                    },
+                    shift: (x, y)=> {
+                        const box = bar._text.getBBox();
+
+                        bar._text._lastCoords = { x: x - box.x, y: y - box.y };
+                        bar._text.attr({ translateX: x - box.x, translateY: y - box.y });
+                    },
+                    draw: ()=> bar.hideLabel(),
+                    getData: ()=> { return { value: bar.getValue() }; },
+                    hideInsideLabel: ()=> false,
+                }]
+            }
+            );
+        return r;
+    }, { left: [], right: [] });
+}
+
+export const clearLabelsCrossTitle = (bars, minY)=> {
+    bars.forEach(bar => {
+        const box = bar._text.getBBox();
+        const lastCoords = bar._text._lastCoords;
+        if(minY > box.y + lastCoords.y) {
+
+            bar.hideLabel();
+        }
+    });
+};
+
+const getStartCoordsArc = function(x, y, innerR, outerR, startAngleCos, startAngleSin, endAngleCos, endAngleSin, isCircle, longFlag) {
```

Comment:
Oh, so many parameters. Shouldn't we pass a configuration object instead of this? https://refactoring.guru/introduce-parameter-object

Reply:
fixed